packages/devextreme/js/__internal/grids/grid_core/header_filter/m_header_filter.ts

```diff
@@ -419,6 +419,12 @@ const ColumnHeadersViewHeaderFilterExtender = extend({}, headerFilterMixin, {
   _columnOptionChanged(e) {
     const { optionNames } = e;
 
+    if (gridCoreUtils.checkChanges(optionNames, ['filterValues', 'filterValue'])) {
```

Comment:
Let's merge this condition with the condition below.
You have two different ```if``` conditions that have almost identical code inside the brackets.

I suggest refactoring the code with the following ideas:

- make a conditions a separate ```bool``` variables
- write one ```if``` related to ```_updateHeaderFilterIndicators```
- wrap ```this.callBase``` into the second ```if```